### PR TITLE
[test] Fix custom decorator handling

### PR DESCRIPTION
**Pull request by gamesh411**
Custom decorators should use functional.wraps for nose test runner to
detect decorated tests. CTU related decorators are extracted, removing
duplication in test suites ctu and ctu_failure.

### CODE REVIEW

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- analyzer/tests/libtest/decorators.py  2
         

Clones removed
==============
+ analyzer/tests/functional/ctu/test_ctu.py  -1
+ analyzer/tests/functional/ctu_failure/test_ctu_failure.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5933836&bid=19445748)

**Comment by gamesh411** [comment]
Also note that ctu_failure had to be refactored, as if a test was skipped, teardown would reference member variables that would otherwise be set only if they were not skipped. Also note that the working directory for commands to run was changed in 2 places, as the original location was not available there. Still, the output of the commands should not depend on the specific directory anyway.

File: analyzer/tests/libtest/decorators.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        def wrapper(self, *args, **kwargs):
+            if not getattr(self, attribute):
+                self.skipTest(message)
+            else:
+                original(self, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+makeSkipUnlessCTUCapable = partial(makeSkipUnlessAttributeFound,
```
Comment (gyorb) [inline analyzer/tests/libtest/decorators.py:32]: These decorators are very CTU specific, what do you think about renaming the module to `ctu_decorators`?

For the general `makeSkipUnlessAttributeFound` a separate decorators could be created if it is used in other tests but I see it only in the ctu related tests now. @csordasmarton what do you think?

Comment (gamesh411) [reply analyzer/tests/libtest/decorators.py]: I have introduced generic decorators and a ctu_decorators module, maybe this improves reusability :)

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- analyzer/tests/libtest/decorators.py  2
         

Clones removed
==============
+ analyzer/tests/functional/ctu/test_ctu.py  -1
+ analyzer/tests/functional/ctu_failure/test_ctu_failure.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5933836&bid=19445748)

**Review by gyorb — APPROVED** [review]

**Pull request merged**